SECURITY.md

```diff
@@ -53,6 +53,88 @@ Here is the security disclosure policy for Node.js
   the release process above to ensure that the disclosure is handled in a
   consistent manner.
 
+## The Node.js threat model
+
+Being able to cause a negative outcome in a way that requires control of the
+elements that are trusted by Node.js is not considered a vulnerability in
+Node.js.
+
+Being able to cause the following through control of the elements that Node.js
+does not trust is considered a vulnerability:
+
+* Disclosure or loss of integrity or confidentiality of data protected through
+  the correct use of Node.js APIs.
+* The unavailability of the runtime, including the unbounded degradation of its
+  performance.
+
+If Node.js loads configuration files or runs code by default (without a
+specific request from the user), and this is not documented, it is considered a
+vulnerability.
+Vulnerabilities related to this case may be fixed by a documentation update.
+
+**Node.js does NOT trust**
+
+1. The data from network connections that are created through the use of Node.js
+   APIs and which is transformed/validated by Node.js before being passed to the
+   application. This includes:
+   * HTTP APIs (all flavors) client and server APIs.
+   * DNS APIs.
+2. Consumers of data protected through the use of Node.js APIs (for example
+   people who have access to data encrypted through the Node.js crypto APIs).
+3. The file content or other I/O that is opened for reading or writing by the
+   use of Node.js APIs (ex: stdin, stdout, stderr). In other words, if the
+   content passing through Node.js to/from the application can trigger actions
+   other than those documented for the APIs, the content is not trusted avoid
+   causing those actions.
```

Comment:
This last sentence is not clear.

Reply:
Yeah. We've been struggling with that. The idea is to say that we don't interpret content when reading files or any I/O such as buffer/stdin/stderr...

Reply:
I'm still wondering if we should remove 3 from the list. I think we trust the application to manage the content that it passes to the APIs. For example if they ask us to read a file that is so big that we OOM then that's on the application.

Maybe we could instead add this line after the "If Node.js loads configuration files...."

If Node.js runs code due to the invocation of an API that does not document that code will be executed when the API is invoked, it is considered a vulnerability.

Reply:
What do you think @mcollina ?

Reply:
The following part is not clear to me:

> the content is not trusted avoid causing those actions

I would break the last sentence out of the bullet point and rewrite as follows:

> In other words, if the data passing through Node.js to/from the application can trigger actions other than those documented for the APIs, there is likely a security vulnerability. Examples of unwanted actions are polluting globals, causing an unrecoverable crash, or any other unexpected side effects that can lead to a loss of confidentiality, integrity, or availability.

Reply:
Thanks for the suggestion. Applied.